staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -449,11 +455,8 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 		server, requestGetter := newHTTP2ServerWithClient(handler, requestTimeout*2)
 		defer server.Close()
 
-		go hardStop(t, stopCh, requestTimeout*3, func() {
-			// the client timed out unexpectedly, let's clean up and abort.
-			close(reqHandlerCompletedCh)
-		})
-
+		// send a request synchronously with a client timeout of requestTimeout*2 seconds
```

Comment:
I'm assuming this removal is a cleanup - not an actual fix, right?

If it is actual fix, can you clarify what was wrong before?

Reply:
it's a bit of both really:
- `hardStop` calls `t.Fatalf` from a goroutine created during the test, this triggers the data race, below is one of the writes that contributed to the data race 
- we don't need `hardStop` really, the client is wired with a timeout, so in case the httptest Server does not come up the request will time out and the test will call `t.Fatalf` from inside the the goroutine that is running the test.


```
Previous write at 0x00c0025d92a1 by goroutine 428:
  testing.(*common).FailNow()
      /usr/local/go/src/testing/testing.go:740 +0x4f
  testing.(*common).Fatalf()
      /usr/local/go/src/testing/testing.go:815 +0x9d
  k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.hardStop()
      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go:1011 +0x1a4
```

Reply:
Yeah - I actually realized that later too (just forgot to remove this comment). Makes sense.